feat: add HTTP API list to reference (#1345)
GreptimeTeam/docs: docs/reference/http-endpoints.md

### PromQL API

- **Path**: `/v1/promql`
- **Methods**: `GET`, `POST`
- **Description**: Executes PromQL queries for Prometheus-compatible metrics.
- **Usage**:
  - `GET`: Send PromQL queries via query parameters.
  - `POST`: Send PromQL queries in the request body.

## Protocol Endpoints

Endpoints for various protocols that are compatible with GreptimeDB. Like InfluxDB, Prometheus, OpenTelemetry, etc.

### InfluxDB Compatibility

- **Paths**:
  - `/v1/influxdb/write`
  - `/v1/influxdb/api/v2/write`
  - `/v1/influxdb/ping`
  - `/v1/influxdb/health`
- **Methods**:
  - `POST` for write endpoints.
  - `GET` for ping and health endpoints.
- **Description**: Provides endpoints compatible with InfluxDB for data ingestion and health checks.
- **Usage**:
  - Ingest data using InfluxDB line protocol.
  - Use ping and health endpoints to check server status.

### Prometheus Remote Write/Read

- **Paths**:
  - `/v1/prometheus/write`
  - `/v1/prometheus/read`
- **Methods**: `POST`
- **Description**: Supports Prometheus remote write and read APIs.
- **Usage**:
  - Send metric data using Prometheus remote write protocol.
  - Read metric data using Prometheus remote read protocol.

> ### Prometheus HTTP API

Author: This is not only for query. This group of endpoints is aimed to provide prometheus-compatible HTTP API as documented here https://prometheus.io/docs/prometheus/latest/querying/api/

> # HTTP API endpoint list

Reviewer: ```suggestion
# HTTP API Endpoint List
```

Here is the full list for the various HTTP paths and their usage in GreptimeDB:

## Miscellaneous

> Endpoints that is not versioned (under `/v1`). For miscellaneous usage like health check, status, metrics, etc.

Author: ```suggestion
Endpoints that is not versioned (under `/v1`). For admin usage like health check, status, metrics, etc.
```

### Status

- **Path**: `/status`
- **Methods**: `GET`
- **Description**: Retrieves the current status of the server.
- **Usage**: Use this endpoint to obtain server status information.

### Metrics

- **Path**: `/metrics`
- **Methods**: `GET`
- **Description**: Exposes Prometheus metrics for monitoring purposes.
- **Usage**: Prometheus can scrape this endpoint to collect metrics data.

### Configuration

- **Path**: `/config`
- **Methods**: `GET`
- **Description**: Retrieves the server's configuration options.
- **Usage**: Access this endpoint to get configuration details.

### Dashboard

- **Paths**: `/dashboard`
- **Methods**: `GET`, `POST`
- **Description**: Provides access to the server's dashboard interface.
- **Usage**: Access these endpoints to interact with the web-based dashboard.

This dashboard is packaged with the GreptimeDB server and provides a user-friendly interface for interacting with the server. It requires corresponding compile flags to be enabled when building GreptimeDB. The original source code for the dashboard can be found at https://github.com/GreptimeTeam/dashboard

### Debugging Tools

- **Base Path**: `/debug`
- **Endpoints**:
  - `/log_level`

>   - `/pprof/*`

Reviewer: What is `*` in `/pprof/*`? Please write in detail.

- **Methods**:
  - `POST` for changing log levels.
  - `GET` for profiling endpoints.
- **Description**: Offers debugging and profiling tools.
- **Usage**:
  - Use `/log_level` to adjust the server's log level dynamically.
  - Use `/pprof/*` endpoints for performance profiling.

## Query Endpoints

Various query APIs for sending query to GreptimeDB.

### SQL API

- **Path**: `/v1/sql`
- **Methods**: `GET`, `POST`
- **Description**: Executes SQL queries against the server.
- **Usage**:

>   - `GET`: Send SQL queries via query parameters.

Reviewer: ```suggestion
```
As discussed before, We do not recommend use the GET method to query data. All SQL API should use `POST`, which means post SQL statements according to the RESTful Principles.

  - `POST`: Send SQL queries in the request body.
  

> For more information on the SQL API, refer to the [SQL documentation](/user-guide/query-data/sql.md).

Reviewer: ```suggestion
For more information on the SQL API, refer to the [HTTP API documentation](/user-guide/protocols/http.md#post-sql-statements) in the user guide.
```
